src/tar.ts

```diff
@@ -5,78 +5,80 @@ import {touch} from './util'
 
 const debug = require('debug')('oclif-update')
 
-export async function extract(stream: NodeJS.ReadableStream, basename: string, output: string, sha?: string) {
-  const getTmp = () => `${output}.partial.${Math.random().toString().split('.')[1].slice(0, 5)}`
-  let tmp = getTmp()
-  if (fs.pathExistsSync(tmp)) tmp = getTmp()
-  debug(`extracting to ${tmp}`)
-  try {
-    await new Promise((resolve, reject) => {
-      const zlib = require('zlib')
-      const tar = require('tar-fs')
-      const crypto = require('crypto')
-      let shaValidated = false
-      let extracted = false
-      const check = () => shaValidated && extracted && resolve()
+export class Tar {
+  static async extract(stream: NodeJS.ReadableStream, basename: string, output: string, sha?: string) {
```

Comment:
What's the point of making this a class?
Functions are simpler and easier to unit test than classes.

Reply:
The function can be stubbed by itself without needing to make it a method in a class though. Unnecessary classes just muddle things up.

I whipped up a bare-bones example of stubbing a function that should be very similar to what you would need to do here at https://github.com/RodEsp/sinon-function-stub-example. Check it out and lemme know if that makes sense.

Reply:
@RodEsp I like your example. It seems to work with default export. How would this work when the exported function is not the default?

Reply:
You'd need to stub the named export then, not the default.

Reply:
@RodEsp thanks, I have made the requested changes